File: programming files/rgb_to_hex.py
# Broken by charlie

# takes in red, green, and blue values and returns a hex string
def rgb_to_hex(r, g, b):
    
    # get rgb values in hex form
    # The 'max' should be 'min'
    r = max(0, max(255, r))
    g = max(0, min(255, g))
    
    #g is supposed to be r
    b = max(0, min(255, g))
    
    # return hex string
    #g and r are swapped
    return '{:02X}{:02X}{:02X}'.format(g, r, b)


# test with hex_color = rgb_to_hex(255, 127, 0) # returns "FF7F00"

File: programming files/rgb_to_hex.java
public class rgb_to_hex {
    public static void main(String[] args) {
        int r = 255;
        int g = 127;
        int b = 0;
        String hexColor = rgbToHex(r, g, b);
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor);
    }

    public static String rgbToHex(int r, int g, int b) {
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.max(0, g));
        b = Math.min(255, Math.max(0, b));
        return String.format("%02X%02X%02X", r, g, b);
    }
}

//Test with RGB color (255, 127, 0) = FF7F00


